keps/sig-instrumentation/2305-metrics-cardinality-enforcement/README.md

```diff
@@ -236,19 +264,145 @@ This would then be interpreted by our machinery as this:
+  feature, can it break the existing applications?).
+  No. The metric cardinality enforcement will be enabled by default.
+  
+* **What happens if we reenable the feature if it was previously rolled back?**
+  There is no functionality concern about enable/disable of the feature.
+  
+* **Are there any tests for feature enablement/disablement?**
+  No.  
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g., what if some components will restart
+   mid-rollout?
+  No known rollout concern.
+* **What specific metrics should inform a rollback?**
+  None.
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and can't do that now.
+  No.
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  A component metric flag for ingesting allowlist to be added.
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  The out-of-bound data will be recorded with label "unexpected" rather than the specific value.
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  None.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  None.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+  None.
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No.
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+  No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  Describe them, providing:
+  No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+  No.
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+  No.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+  No.
```

Comment:
This might, since we have to do runtime checks for emitting metrics against an allow-list.

Reply:
It may not be that obvious? But I put the word there. Is there any specific SLI/SLO will be affected?

Reply:
I'm confused by that statement @ehashman.. It may affect the time it takes to populate a metric but that actually shouldn't affect SLOs or SLIs since the metric itself will have correct values.

Reply:
> It may affect the time it takes to populate a metric but that actually shouldn't affect SLOs or SLIs since the metric itself will have correct values.

If operations block on any of this it could increase e.g. the duration of an HTTP call which is being instrumented by a metric, if we messed up the implementation.